app/models/course_queue_entry.rb

```diff
@@ -1,11 +1,12 @@
 class CourseQueueEntry < ApplicationRecord
   belongs_to :course_queue
   has_one :course, through: :course_queue
-  belongs_to :requester, class_name: 'User'
+  belongs_to :course_group, required: false
+  belongs_to :requester, class_name: 'User', required: false
   belongs_to :resolver, class_name: 'User', required: false
 
-  def is_requester?(user)
-    user == self.requester
+  def is_requester?(user_or_group)
+    user_or_group == self.requester || user_or_group == self.course_group
```

Comment:
Where is this method called? Seems like it could be used in `current_user_only`.